src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -247,32 +242,28 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
   if (!logic_uuid.empty() && !boost::filesystem::exists(xclbinPath)) {
     logger(_ptTest, "Details", "Verify xclbin not available or shell partition is not programmed. Skipping validation.");
     _ptTest.put("status", test_token_skipped);
-    return;
+    throw xrt_core::error(std::errc::operation_canceled);
```

Comment:
Thank you the refactoring an throwing an exception (instead of simply returning from the method).